### PR TITLE
Run commands in their own sessions

### PR DESCRIPTION
**Pull request by edigaryev**
Should help with the Zsh behavior on FreeBSD/macOS.

### CODE REVIEW

File: internal/executor/shell_unix_test.go [hunk 1]
```diff
@@ -8,6 +8,10 @@ import (
 	"time"
 )
 
+func TestZshDoesNotHang(t *testing.T) {
+	_, _ = ShellCommandsAndGetOutput([]string{"zsh -c 'echo \"a:b\" | read -d \":\" piece'"}, nil, nil)
```
Comment (fkorotkov) [inline internal/executor/shell_unix_test.go:12]: Let's either add `timeout` here or to the Cirrus task so in case of hanging it won't hang for an hour

**Review by fkorotkov — DISMISSED** [review]

File: internal/executor/shell_unix_test.go (hunk 1 above)
Comment (edigaryev) [reply internal/executor/shell_unix_test.go]: Added a timeout in https://github.com/cirruslabs/cirrus-ci-agent/pull/98/commits/d41b9382b190b7dc7f4368d7710cee626ba8b2e0.

**Review by fkorotkov — APPROVED** [review]

File: .cirrus.yml [hunk 2]
```diff
@@ -21,9 +21,10 @@ task:
   modules_cache:
     fingerprint_script: cat go.sum
     folder: $GOPATH/pkg/mod
-  get_script: go get ./...
-  build_script: go build ./...
-  test_script: go test ./...
+  test_script:
+    - apt-get update
+    - apt-get install -y zsh
```
Comment (fkorotkov) [inline .cirrus.yml:26]: should we just test it on FreeBSD like in #99? Anyways the issue was raised by a FreeBSD user

Comment (edigaryev) [reply .cirrus.yml:26]: Why not test it on Linux too? It's almost free.

Comment (edigaryev) [reply .cirrus.yml:26]: Also a better test case wouldn't require Zsh at all, but I don't have any ATM.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**